internal/rootcoord/meta_table.go

```diff
@@ -83,18 +89,40 @@ type MetaTable struct {
 	credLock sync.RWMutex
 }
 
+// TODO using factory once dependency of tnx and snap are removed
+func newCatalog(txn kv.TxnKV, snap kv.SnapShotKV) (metastore.Catalog, error) {
```

Comment:
should only  kv catalog use txn and snap

Reply:
Ok, will introduce a factory as TODO says.